### PR TITLE
Feature/stock photos editor and analytics

### LINKED ISSUE #7311: Stock photos: support stock photos in media toolbar
Add stock photos as a choice from the editor's "Add Media" feature (design forthcoming).

### LINKED ISSUE #7315: Stock photos: insert into post
Insert selected stock photo into the current post after upload completes.

### LINKED ISSUE #7501: Stock photos: analytics
Before the stock photos master branch is merged into develop, we should add analytics to capture the following:

- When stock photos is accessed
- When the user performs a stock photo search
- When - and how many - stock photos are added (uploaded) to the user's media library
- When - and how many - stock photos are added to a post

### PR DESCRIPTION
**Pull request by nbradbury**
Resolves #7315, resolves #7311, resolves #7501 - This is final stock photos PR before design review, which adds simple analytics and integration with the editor.

To test:

- Open the editor (Aztec only)
- Tap to add media
- Tap the gallery icon
- Choose "Free Photo Library"
- Search stock photos
- Pick a photo
- Tap Add

The photo should be added to the editor.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java [hunk 1]
```diff
@@ -395,12 +398,18 @@ public void onStockMediaUploaded(MediaStore.OnStockMediaUploaded event) {
...
-                idList.add(media.getId());
+            int count = event.mediaList.size();
+            long[] idArray = new long[count];
+            for (int i = 0; i < count; i++) {
+                idArray[i] = event.mediaList.get(i).getMediaId();
             }
+
+            Map<String, Integer> properties = new HashMap<>();
+            properties.put("count", count);
+            AnalyticsTracker.track(AnalyticsTracker.Stat.STOCK_MEDIA_UPLOADED, properties);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java]: I wonder if we should reply here the approach followed when bumping analytics for other photos. 
Analytics are bumped for each picture on its own call, by adding properties of the media to the event. See `trackAddRecentMediaEvent` in `PhotoPickerFragment` and how we used `AnalyticsUtils.getMediaProperties` to attach properties of the media to the event.

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java [hunk 2]
```diff
@@ -962,6 +967,12 @@ public static String getEventNameForStat(AnalyticsTracker.Stat stat) {
                 return "plugin_search_performed";
             case PLUGIN_UPDATED:
                 return "plugin_updated";
+            case STOCK_MEDIA_ACCESSED:
+                return "stock_media_accessed";
+            case STOCK_MEDIA_SEARCHED :
+                return "stock_media_searched";
+            case STOCK_MEDIA_UPLOADED:
+                return "stock_media_uploaed";
```
Comment (daniloercoli) [inline libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java]: There is a typo here.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 3]
```diff
@@ -2492,6 +2496,24 @@ public void done() {
...
+
+                        Map<String, Object> properties = new HashMap<>();
+                        properties.put("count", mediaIds.length);
+                        AnalyticsUtils.trackWithSiteDetails(
+                                Stat.EDITOR_ADDED_PHOTO_VIA_STOCK_MEDIA_LIBRARY,
+                                mSite,
+                                properties);
+
+                        for (long id : mediaIds) {
+                            addExistingMediaToEditor(id);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: I noticed that `addExistingMediaToEditor` already bumps analytics when it calls `trackAddMediaFromWPLibraryEvents` (aka EDITOR_ADDED_VIDEO_VIA_WP_MEDIA_LIBRARY). 

Is it on purpose?

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java (hunk 2 above)
Comment (nbradbury) [reply libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java]: Whoops, fixed in d342ab7

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 3 above)
Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: > I noticed that addExistingMediaToEditor already bumps analytics when it calls trackAddMediaFromWPLibraryEvents

I totally missed that - thanks for catching it. I resolved this by adding a "source" parameter to the tracking method [here](https://github.com/wordpress-mobile/WordPress-Android/blob/975837e5f013fc6dda20d36ae7d7a2d615b9dd1a/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L2214).

Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: 👍

**Comment by nbradbury** [comment]
Thanks for tackling this @daniloercoli. I believe I've addressed the issues you raised, so ready for another round.

File: WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java [hunk 4]
```diff
@@ -395,17 +400,43 @@ public void onStockMediaUploaded(MediaStore.OnStockMediaUploaded event) {
...
+            AppLog.e(AppLog.T.MEDIA, "Cannot track uploaded stock media event if mediaList is empty");
+            return;
+        }
+
+        boolean isMultiselect = mediaList.size() > 1;
+
+        for (MediaModel media : mediaList) {
+            if (media.getUrl() != null) {
+                Uri mediaUri = Uri.parse(media.getUrl());
+                Map<String, Object> properties =
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java]: Sorry Nick, I suggest you to track media details but noticed that our utility routine doesn't work for remote files. I suggest to remove this call here, and create just an empty properties Map. Maybe adding a TODO to improve the routine to take in consideration remote files as well as local files.

Other than that I think it's ok and ready to be merged.

**Comment by daniloercoli** [comment]
Looks good! :shipit:

**Pull request merged**